### PR TITLE
Clean up dead code from service mesh changes

### PR DESCRIPTION
**Pull request by wangyyt1013**
Signed-off-by: Yuanyuting Wang <yuanyutingwang@fb.com>

## Summary

- Remove `Indexer.GetIndexerInfo` rpc and related objects / function implementations
- Remove `reporter_hwid` message field from `IndexRequest` proto
- Remove `StreamProvider.GetStreamName` function definition and related implementations, and adjust related test cases to bypass this change

## Test Plan

Run all existing test cases; all tests should pass

### CODE REVIEW

File: feg/gateway/policydb/policydb_test.go [hunk 1]
```diff
@@ -181,7 +177,7 @@ func TestPolicyDBBaseNamesWithGRPC(t *testing.T) {
 
 func TestPolicyDBRulesWithGRPC(t *testing.T) {
 	onceTestsInit.Do(func() { initOnce(t) })
-	streamer_test_init.StartNewTestProvider(t, &mockStreamProvider{Name: "policydb"})
+	streamer_test_init.StartNewTestProvider(t, &mockStreamProvider{Name: "policydb"}, "policydb")
```
Comment (hcgatewood) [inline feg/gateway/policydb/policydb_test.go]: `mockStreamProvider` doesn't need to know what its stream name is, so can remove it altogether from the object defn

Comment (hcgatewood) [inline feg/gateway/policydb/policydb_test.go]: `mockStreamProvider` doesn't need to know what its stream name is, so can remove it altogether from the object defn

File: orc8r/cloud/go/services/streamer/integ_test.go [hunk 2]
```diff
@@ -117,7 +113,7 @@ func TestStreamerClient(t *testing.T) {
 
 	streamerClient := streamer_client.NewStreamerClient(mockedCloudRegistry{})
 	mockProvider := &mockStreamProvider{name: testStreamName, retVal: expected}
-	streamer_test_init.StartNewTestProvider(t, mockProvider)
+	streamer_test_init.StartNewTestProvider(t, mockProvider, mockProvider.name)
```
Comment (hcgatewood) [inline orc8r/cloud/go/services/streamer/integ_test.go]: Related to my above comment: this should just be `t, mockProvider, testStreamName`

Comment (hcgatewood) [inline orc8r/cloud/go/services/streamer/integ_test.go]: Related to my above comment: this should just be `t, mockProvider, testStreamName`

File: orc8r/cloud/go/services/streamer/test_init/test_provider_servicer.go [hunk 3]
```diff
@@ -30,14 +30,14 @@ type providerServicer struct {
 
 // StartNewTestProvider starts a new stream provider service which forwards
 // calls to the passed provider.
-func StartNewTestProvider(t *testing.T, provider providers.StreamProvider) {
+func StartNewTestProvider(t *testing.T, provider providers.StreamProvider, name string) {
```
Comment (hcgatewood) [inline orc8r/cloud/go/services/streamer/test_init/test_provider_servicer.go]:
```suggestion
func StartNewTestProvider(t *testing.T, provider providers.StreamProvider, streamName string) {
```

Comment (hcgatewood) [inline orc8r/cloud/go/services/streamer/test_init/test_provider_servicer.go]:
```suggestion
func StartNewTestProvider(t *testing.T, provider providers.StreamProvider, streamName string) {
```

**Review by hcgatewood — CHANGES_REQUESTED** [review]
Looks great, thanks Elaine. Just one small cleanup then we can get it merged

**Review by themarwhal — APPROVED** [review]
The FeG changes lgtm! :)

**Comment by wangyyt1013** [comment]
Creating [new PR](https://github.com/magma/magma/pull/7146) for merging based on changed made in this PR.

**Pull request closed without merging**